### DISCUSSION TITLE
Hardware Requirements for Smokeping Master-Slave Setup Monitoring 4000 Devices

### DISCUSSION THREAD

**Discussion by tijender-Kumar** [oetiker/SmokePing#422] [General]
I am planning to set up Smokeping with a master-slave architecture to monitor a network of 4000 devices. The setup will consist of:

1 Master and 4 Slaves
Probing will use fping with 10 probes every 60 seconds.
I’d like to ensure the system runs smoothly with minimal latency and resource issues. Can someone recommend the appropriate hardware specifications (CPU, RAM, Disk space, etc.) for both the master and slave servers to handle this setup? Additionally, what are the network bandwidth requirements for communication between the master and slaves?

Any guidance or experience with similar setups would be greatly appreciated.

**Comment by moetiker** [comment]
Hi,

I think you do not have to worry about resources ... smokeping and fping do need much ... 16 GB Ram and an 4 Core CPU will Handel that. The disk space depends on your rrd configuration ... but more the 30 GB (4kx2MB) is probably not needed to hold all rrd's ...  (I have 1000 rrd's and it needs 1.2GB)

Manuel

**Reply by tijender-Kumar** [reply]
Can you please explain the configuration logic for 16GB RAM and 4 core CPU, I have master located in the cloud with nagios icinga configuration and smokeping as master running as extra service on it and all 4 slaves are local with 8 plus GB RAM.

I was hoping maybe I don't need much resources on the cloud based master as all probing is done on local slaves. But as I am getting OOM killer message for the nagios icinga service intermittently.

Hence I am more curious to know what might be causing it. Therefore any insight why smokeping master require high resources will be much appreciated.

**Comment by moetiker** [comment]
oh, I thought you're going with HW... if you have a cloud service just try with less and give the instance more .. if it needs more ...
start with 2gb ram and 1 cpu ...

**Reply by tijender-Kumar** [reply]
Thanks, But I am working for a small ISP which have limited resources and thus we need to give explanation for every GIG that need to be added to cloud instance for cost reduction purpose. So if I can have a explanation for supporting the HARDWARE requirement I will be in better position to get approval from my Clients.
